src/main/java/uk/gov/hmcts/darts/arm/component/AutomatedTaskProcessorFactory.java

```diff
@@ -14,4 +15,6 @@ public interface AutomatedTaskProcessorFactory {
     GenerateCaseDocumentProcessor createGenerateCaseDocumentProcessor(int batchSize);
 
     CleanupCurrentFlagEventProcessor createCleanupCurrentFlagEventProcessor(int batchSize);
+
+    GenerateCaseDocumentForRetentionDateProcessor createGenerateCaseDocumentForRetentionDate(int batchSize);
```

Comment:
shall we follow existing naming conventions?
```suggestion
    GenerateCaseDocumentForRetentionDateProcessor createGenerateCaseDocumentForRetentionDateProcessor(int batchSize);
```